### PR TITLE
Update Developer Guide's Model diagram

### LINKED ISSUE #181: Update Model UML diagram in DG

### PR DESCRIPTION
**Pull request by PresleyChew**
Fixes #181

### CODE REVIEW

**Review by yong-zhuo — APPROVED** [review]

File: docs/images/BetterModelClassDiagram.png
Comment (yadobler) [inline docs/images/BetterModelClassDiagram.png:1]: wait this is not the correct image, is it?

**Review by yadobler — CHANGES_REQUESTED** [review]

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -9,10 +9,10 @@ AddressBook *-right-> "1" UniqueTagList
 UniqueTagList -[hidden]down- UniquePersonList
 UniqueTagList -[hidden]down- UniquePersonList
 
-UniqueTagList -right-> "*" Tag
+UniqueTagList -right-> "*" Allergy
```
Comment (yong-zhuo) [inline docs/diagrams/BetterModelClassDiagram.puml:12]: Have to change to `UniqueAllergyList`

File: docs/images/BetterModelClassDiagram.png
Comment (yong-zhuo) [reply docs/images/BetterModelClassDiagram.png:1]: I don't think he generated the .png file for this .puml diagram yet

**Review by yong-zhuo — CHANGES_REQUESTED** [review]

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 1 above)
Comment (yadobler) [reply docs/diagrams/BetterModelClassDiagram.puml:12]: Can add one for `OwnedAppointmentList` also

**Review by yong-zhuo — APPROVED** [review]

**Review by yadobler — APPROVED** [review]

File: docs/img.png
Comment (yadobler) [inline docs/img.png:1]: lmao this must not be in release

**Review by yadobler — APPROVED** [review]

**Pull request merged**